great_expectations/data_context/data_context/abstract_data_context.py

```diff
@@ -267,6 +290,49 @@ def config_variables(self) -> Dict:
             self._config_variables = self._load_config_variables()
         return self._config_variables
 
+    @property
+    def config(self) -> DataContextConfig:
+        return self._project_config
+
+    @property
+    def root_directory(self) -> Optional[str]:
+        """The root directory for configuration objects in the data context; the location in which
+        ``great_expectations.yml`` is located.
+
+        Why does this exist in AbstractDataContext? CloudDataContext and FileDataContext both use it
+
+        """
+        if hasattr(self, "_context_root_directory"):
+            return self._context_root_directory
+        return
+
+    @property
+    def project_config_with_variables_substituted(self) -> DataContextConfig:
+        return self.get_config_with_variables_substituted()
+
+    @property
+    def plugins_directory(self):
+        """The directory in which custom plugin modules should be placed.
+
+        Why does this exist in AbstractDataContext? CloudDataContext and FileDataContext both use it
```

Comment:
Since `CloudDataContext` currently references the filesystem, how do we feel about inheriting from `FileDataContext` instead of `AbstractDataContext` e.g. `CloudDataContext(FileDataContext)`? That might allow us to remove the file based methods/attributes from `AbstractDataContext`.

Reply:
Alternatively, we could refactor the file interaction code into a separate class that is injected as a dependency where we need it (File and Cloud).

Reply:
@anthonyburdi this is a great idea, and I think it will definitely make things simpler going forward. I can implement the adjustment to the hierarchy between `CloudDataContext` and `FileDataContext` in this PR and will make a note of the possibility of the file interaction code into separate class. 

Thanks for the synchronous conversation as well

Reply:
update: after a conversation with @donaldheppner this understanding has been updated: 
- The key difference between *`DataContext` is going to be where the initial configuration comes from : 
- `CloudDataContext` loads config from endpoint
- `FileDataContext` loads `great_expectations.yml` from disk
-  `EphemeralDataContext` doesn't have a yml at all, but loads from in-memory dictionary. 

Afterwards a lot of the behavior is going to be shared, which means all DataContexts will have a way to access the file system, config files, Stores on filesystem, etc (which is what GE currently supports). 

This means a LOT of the functionality is going to be living in AbstractDataContext, and `Cloud` and `File` will remain as siblings.